Security Certificate Tests fail in JDK 1.8.0_292 due to JDK bug

I (@tvernum) have renamed & updated this issue to be an umbrella issue for all occurrences of this particular problem.

JDK 1.8.0_292 is affected by https://bugs.openjdk.java.net/browse/JDK-8266279 which sometimes causes failures like:

```
 java.security.KeyStoreException: Key protection  algorithm not found: java.security.UnrecoverableKeyException: Encrypt Private Key failed: unrecognized algorithm name: PBEWithSHA1AndDESede
```

Some of our tests that deal with X.509 certificates (e.g. TLS & SAML signing test cases) are failing due to this error.
 
Since this is a JDK bug that affects a specific build, is scheduled to be fixed in an upcoming release, we are actively muting the affect tests for that specific JDK build with:

```
        assumeFalse("JDK bug JDK-8266279, https://github.com/elastic/elasticsearch/issues/75571",
            "1.8.0_292".equals(System.getProperty("java.version")));
```

We're linking all of them to this issue so that we can find & handle all such test-mutes in bulk when the next JDK build is added to CI.

Because this is an issue for JDK8, it only affects 7.x and 6.8. We do not run CI for the `master` (8.0) branch on JDK8.

**This was the original issue description**

-----
Looks like the problem will be fixed in latest jdk update
https://bugs.openjdk.java.net/browse/JDK-8266279
fails on jdk8.u292 in CI

**Build scan**:
https://gradle-enterprise.elastic.co/s/5s657scdqzsz2
**Repro line**:
./gradlew ':x-pack:plugin:security:unitTest' -Dtests.seed=9915999ED6C30B91 -Dtests.class=org.elasticsearch.xpack.security.authc.saml.SamlRealmTests -Dtests.method="testCreateSigningCredentialFromKeyStoreFailureScenarios" -Dtests.security.manager=true -Dtests.locale=ar-DZ -Dtests.timezone=Asia/Anadyr -Dcompiler.java=11 -Druntime.java=8
**Reproduces locally?**:
no
passes locally on adoptopenjdk 1.8.0_275
and passes locally on AdoptOpenJDK 1.8.0_292 [OpenJDK 64-Bit Server VM 25.292-b10]

**Applicable branches**:
6.8
**Failure history**:
https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-30d,mode:quick,to:now))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:b646ed00-7efc-11e8-bf69-63c8ef516157,key:process.status,negate:!f,params:(query:FAILURE,type:phrase),type:phrase,value:FAILURE),query:(match:(process.status:(query:FAILURE,type:phrase))))),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:testCreateSigningCredentialFromKeyStoreFailureScenarios),sort:!(process.time-start,desc))

**Failure excerpt**:
| SamlRealmTests.testCreateEncryptionCredentialFromKeyStore <<< FAILURES!
10:56:41    > Throwable #1: java.security.KeyStoreException: Key protection  algorithm not found: java.security.UnrecoverableKeyException: Encrypt Private Key failed: unrecognized algorithm name: PBEWithSHA1AndDESede
10:56:41    > 	at __randomizedtesting.SeedInfo.seed([9915999ED6C30B91:CD603BE1C03300AB]:0)
10:56:41    > 	at sun.security.pkcs12.PKCS12KeyStore.setKeyEntry(PKCS12KeyStore.java:694)
10:56:41    > 	at sun.security.pkcs12.PKCS12KeyStore.engineSetKeyEntry(PKCS12KeyStore.java:594)
10:56:41    > 	at java.security.KeyStore.setKeyEntry(KeyStore.java:1140)
10:56:41    > 	at org.elasticsearch.xpack.security.authc.saml.SamlRealmTests.testCreateEncryptionCredentialFromKeyStore(SamlRealmTests.java:415)
10:56:41    > 	at java.lang.Thread.run(Thread.java:748)
10:56:41    > Caused by: java.security.UnrecoverableKeyException: Encrypt Private Key failed: unrecognized algorithm name: PBEWithSHA1AndDESede
10:56:41    > 	at sun.security.pkcs12.PKCS12KeyStore.encryptPrivateKey(PKCS12KeyStore.java:938)
10:56:41    > 	at sun.security.pkcs12.PKCS12KeyStore.setKeyEntry(PKCS12KeyStore.java:631)
10:56:41    > 	... 40 more
10:56:41    > Caused by: java.security.NoSuchAlgorithmException: unrecognized algorithm name: PBEWithSHA1AndDESede
10:56:41    > 	at sun.security.x509.AlgorithmId.get(AlgorithmId.java:448)
10:56:41    > 	at sun.security.pkcs12.PKCS12KeyStore.mapPBEAlgorithmToOID(PKCS12KeyStore.java:955)
10:56:41    > 	at sun.security.pkcs12.PKCS12KeyStore.encryptPrivateKey(PKCS12KeyStore.java:912)

--- a/x-pack/plugin/security/cli/src/test/java/org/elasticsearch/xpack/security/cli/CertificateGenerateToolTests.java
+++ b/x-pack/plugin/security/cli/src/test/java/org/elasticsearch/xpack/security/cli/CertificateGenerateToolTests.java
@@ -266,10 +266,6 @@ public class CertificateGenerateToolTests extends ESTestCase {
     }
 
     public void testGeneratingSignedCertificates() throws Exception {
-        assumeFalse(
-            "JDK bug JDK-8266279, https://github.com/elastic/elasticsearch/issues/72639",
-            "1.8.0_292".equals(System.getProperty("java.version"))
-        );
 
         Path tempDir = initTempDir();
         Path outputFile = tempDir.resolve("out.zip");

--- a/x-pack/plugin/security/cli/src/test/java/org/elasticsearch/xpack/security/cli/CertificateToolTests.java
+++ b/x-pack/plugin/security/cli/src/test/java/org/elasticsearch/xpack/security/cli/CertificateToolTests.java
@@ -425,10 +425,6 @@ public class CertificateToolTests extends ESTestCase {
     }
 
     public void testHandleLongPasswords() throws Exception {
-        assumeFalse(
-            "JDK bug JDK-8266279, https://github.com/elastic/elasticsearch/issues/75571",
-            "1.8.0_292".equals(System.getProperty("java.version"))
-        );
 
         final Path tempDir = initTempDir();
 
@@ -655,10 +651,6 @@ public class CertificateToolTests extends ESTestCase {
      * - Checks that all 3 certificates have the right values based on the command line options provided during generation
      */
     public void testCreateCaAndMultipleInstances() throws Exception {
-        assumeFalse(
-            "JDK bug JDK-8266279, https://github.com/elastic/elasticsearch/issues/75571",
-            "1.8.0_292".equals(System.getProperty("java.version"))
-        );
         final Path tempDir = initTempDir();
 
         final MockTerminal terminal = new MockTerminal();
@@ -839,10 +831,6 @@ public class CertificateToolTests extends ESTestCase {
      * - Checks that the PKCS12 certificate and the PEM certificate trust one another
      */
     public void testTrustBetweenPEMandPKCS12() throws Exception {
-        assumeFalse(
-            "JDK bug JDK-8266279, https://github.com/elastic/elasticsearch/issues/75571",
-            "1.8.0_292".equals(System.getProperty("java.version"))
-        );
         final Path tempDir = initTempDir();
 
         final MockTerminal terminal = new MockTerminal();

--- a/x-pack/plugin/security/cli/src/test/java/org/elasticsearch/xpack/security/cli/HttpCertificateCommandTests.java
+++ b/x-pack/plugin/security/cli/src/test/java/org/elasticsearch/xpack/security/cli/HttpCertificateCommandTests.java
@@ -118,14 +118,6 @@ public class HttpCertificateCommandTests extends ESTestCase {
         assumeFalse("Can't run in a FIPS JVM", inFipsJvm());
     }
 
-    @BeforeClass
-    public static void muteOnBrokenJdk() {
-        assumeFalse(
-            "JDK bug JDK-8266279, https://github.com/elastic/elasticsearch/issues/72359",
-            "1.8.0_292".equals(System.getProperty("java.version"))
-        );
-    }
-
     public void testGenerateSingleCertificateSigningRequest() throws Exception {
         final Path outFile = testRoot.resolve("csr.zip").toAbsolutePath();
 

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlRealmTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlRealmTests.java
@@ -41,7 +41,6 @@ import org.elasticsearch.xpack.security.authc.Realms;
 import org.elasticsearch.xpack.security.authc.support.MockLookupRealm;
 import org.hamcrest.Matchers;
 import org.junit.Before;
-import org.junit.BeforeClass;
 import org.mockito.Mockito;
 import org.opensaml.saml.common.xml.SAMLConstants;
 import org.opensaml.saml.metadata.resolver.impl.AbstractReloadingMetadataResolver;

--- a/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlRealmTests.java
+++ b/x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/saml/SamlRealmTests.java
@@ -118,14 +118,6 @@ public class SamlRealmTests extends SamlTestCase {
         threadContext = new ThreadContext(globalSettings);
     }
 
-    @BeforeClass
-    public static void muteOnBrokenJdk() {
-        assumeFalse(
-            "JDK bug JDK-8266279, https://github.com/elastic/elasticsearch/issues/75571",
-            "1.8.0_292".equals(System.getProperty("java.version"))
-        );
-    }
-
     public void testReadIdpMetadataFromFile() throws Exception {
         final Path path = getDataPath("idp1.xml");
         Tuple<RealmConfig, SSLService> config = buildConfig(path.toString());

